tsdb/engine/tsm1/file_store.go

```diff
@@ -633,6 +629,43 @@ func (f *FileStore) BlockCount(path string, idx int) int {
 	return 0
 }
 
+// walkFiles calls fn for every files in filestore in parallel
+func (f *FileStore) walkFiles(fn func(f TSMFile) error) error {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
+
+	// struct to hold the result of opening each reader in a goroutine
+	type res struct {
+		err error
+	}
```

Comment:
Seems like you could just do a `chan error` since there's only one field in the struct.